### ISSUE TITLE
cross-staff notes in secondary layers cannot find correct clef

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#241]
When cross-staff notes are present in layers other than the first one, they are displayed on the staff using the clef of the home staff rather than the away staff.

Correct display with only one layer in the top staff:

![screen shot 2016-07-16 at 2 35 10 pm](https://cloud.githubusercontent.com/assets/3487289/16897318/88d67d38-4b62-11e6-90db-730b954880d5.png)

MEI data:

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-07-16 14:35:06</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.12-dev-2657f99</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000073727339">
                        <staffGrp xml:id="m-000000196049415" symbol="brace">
                            <staffDef xml:id="staffdef-000000118373439" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5" />
                            <staffDef xml:id="staffdef-000000183741661" clef.shape="F" clef.line="4" meter.count="4" meter.unit="4" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <measure xml:id="measure-000000189673240" n="1" right="end">
                            <staff xml:id="staff-000000011758963" n="1">
                                <!-- kern: =1-  8eL  8BJj  8ALj  8GJj  8FLj  8AJj  4c  == -->
                                <layer xml:id="layer-000000021871880" n="1">
                                    <beam xml:id="beam-000000166248603">
                                        <note xml:id="note-000000049312635" dur="8" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-000000083527151" dur="8" staff="2" oct="3" pname="b" accid.ges="n" />
                                    </beam>
                                    <beam xml:id="beam-000000030781939">
                                        <note xml:id="note-000000023244933" dur="8" staff="2" oct="3" pname="a" accid.ges="n" />
                                        <note xml:id="note-000000050321945" dur="8" staff="2" oct="3" pname="g" accid.ges="n" />
                                    </beam>
                                    <beam xml:id="beam-000000081872787">
                                        <note xml:id="note-000000143171921" dur="8" staff="2" oct="3" pname="f" accid.ges="n" />
                                        <note xml:id="note-000000035061548" dur="8" staff="2" oct="3" pname="a" accid.ges="n" />
                                    </beam>
                                    <note xml:id="note-000000009931222" dur="4" oct="4" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                            <staff xml:id="staff-000000064398978" n="2">
                                <!-- kern: =1-  1C  == -->
                                <layer xml:id="layer-000000054568782" n="1">
                                    <note xml:id="note-000000162003485" dur="1" oct="3" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Incorrect display of second layer of top staff when another layer is added to the top staff (eighth notes should be the same pitches as before, and they are if you imagine the bottom clef as a treble clef):

![screen shot 2016-07-16 at 2 36 39 pm](https://cloud.githubusercontent.com/assets/3487289/16897326/bdb74c08-4b62-11e6-9856-ae0d4d4db68f.png)

MEI data:

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-07-16 14:32:41</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.12-dev-2657f99</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000019771936">
                        <staffGrp xml:id="m-000000128323375" symbol="brace">
                            <staffDef xml:id="staffdef-000000013152069" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5" />
                            <staffDef xml:id="staffdef-000000158779955" clef.shape="F" clef.line="4" meter.count="4" meter.unit="4" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <measure xml:id="measure-000000103427459" n="1" right="end">
                            <staff xml:id="staff-000000050789578" n="1">
                                <!-- kern: =1-  *^  2r  2cc  *v  == -->
                                <layer xml:id="layer-000000092172722" n="1">
                                    <rest xml:id="rest-000000166995837" dur="2" />
                                    <note xml:id="note-000000152667640" dur="2" oct="5" pname="c" accid.ges="n" />
                                </layer>
                                <!-- kern: 8eL  8BJj  8ALj  8GJj  8FLj  8AJj  4c  *v -->
                                <layer xml:id="layer-000000071569086" n="2">
                                    <beam xml:id="beam-000000056046120">
                                        <note xml:id="note-000000080824711" dur="8" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-000000137517772" dur="8" staff="2" oct="3" pname="b" accid.ges="n" />
                                    </beam>
                                    <beam xml:id="beam-000000139298144">
                                        <note xml:id="note-000000211994378" dur="8" staff="2" oct="3" pname="a" accid.ges="n" />
                                        <note xml:id="note-000000099395669" dur="8" staff="2" oct="3" pname="g" accid.ges="n" />
                                    </beam>
                                    <beam xml:id="beam-000000015491646">
                                        <note xml:id="note-000000093086389" dur="8" staff="2" oct="3" pname="f" accid.ges="n" />
                                        <note xml:id="note-000000061116529" dur="8" staff="2" oct="3" pname="a" accid.ges="n" />
                                    </beam>
                                    <note xml:id="note-000000044076223" dur="4" oct="4" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                            <staff xml:id="staff-000000210447893" n="2">
                                <!-- kern: =1-  1C  == -->
                                <layer xml:id="layer-000000121980194" n="1">
                                    <note xml:id="note-000000133246258" dur="1" oct="3" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Here are the warnings which are related to the problem:

```
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000137517772'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000211994378'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000099395669'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000093086389'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000061116529'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000137517772'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000211994378'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000099395669'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000093086389'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000061116529'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000137517772'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000211994378'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000099395669'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000093086389'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000061116529'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000137517772'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000211994378'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000099395669'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000093086389'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000061116529'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000137517772'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000211994378'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000099395669'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000093086389'
[Warning] Could not get the layer with cross-staff reference '2' for element 'note-000000061116529'
```

The problem is fixed if a second layer is added to the second staff:

![screen shot 2016-07-16 at 2 42 52 pm](https://cloud.githubusercontent.com/assets/3487289/16897357/a5d01d08-4b63-11e6-8457-24b32fcd1c1f.png)

XML data:

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-07-16 14:43:42</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.12-dev-2657f99</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000053947835">
                        <staffGrp xml:id="m-000000200096990" symbol="brace">
                            <staffDef xml:id="staffdef-000000070724770" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5" />
                            <staffDef xml:id="staffdef-000000177785418" clef.shape="F" clef.line="4" meter.count="4" meter.unit="4" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <measure xml:id="measure-000000170764156" n="1" right="end">
                            <staff xml:id="staff-000000102422404" n="1">
                                <!-- kern: =1-  *^  2r  2cc  *v  == -->
                                <layer xml:id="layer-000000196751510" n="1">
                                    <rest xml:id="rest-000000107310600" dur="2" />
                                    <note xml:id="note-000000112489130" dur="2" oct="5" pname="c" accid.ges="n" />
                                </layer>
                                <!-- kern: 8eL/  8BJj/  8ALj/  8GJj/  8FLj  8AJj  4c  *v -->
                                <layer xml:id="layer-000000174955136" n="2">
                                    <beam xml:id="beam-000000136371363">
                                        <note xml:id="note-000000198963308" dur="8" oct="4" pname="e" accid.ges="n" stem.dir="up" />
                                        <note xml:id="note-000000129534173" dur="8" staff="2" oct="3" pname="b" accid.ges="n" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000176684412">
                                        <note xml:id="note-000000209273777" dur="8" staff="2" oct="3" pname="a" accid.ges="n" stem.dir="up" />
                                        <note xml:id="note-000000115654663" dur="8" staff="2" oct="3" pname="g" accid.ges="n" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000120473822">
                                        <note xml:id="note-000000155943962" dur="8" staff="2" oct="3" pname="f" accid.ges="n" />
                                        <note xml:id="note-000000161133258" dur="8" staff="2" oct="3" pname="a" accid.ges="n" />
                                    </beam>
                                    <note xml:id="note-000000189788339" dur="4" oct="4" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                            <staff xml:id="staff-000000205202638" n="2">
                                <!-- kern: =1-  *^  1C  *v  == -->
                                <layer xml:id="layer-000000115152683" n="1">
                                    <note xml:id="note-000000058895630" dur="1" oct="3" pname="c" accid.ges="n" />
                                </layer>
                                <!-- kern: 4GG  2.EE  *v -->
                                <layer xml:id="layer-000000083655634" n="2">
                                    <note xml:id="note-000000042696947" dur="4" oct="2" pname="g" accid.ges="n" />
                                    <note xml:id="note-000000133303447" dots="1" dur="2" oct="2" pname="e" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Therefore, the bug is most likely resulting from searching for a matching layer on the away staff.  The first layer should be searched for instead on the away staff, or at least if a matching layer is not found on the away staff, then search for a lower numbered layer or the first layer.

This bug is probably related to the temporal clef change bug fix, as I don't recall seeing it last week.

**Comment by craigsapp** [comment]
Hey this is still here... Somewhat like issue #400, but not exactly as duplicating a clef is not possible.

Here is another test file:

<img width="773" alt="screen shot 2016-12-28 at 17 58 45" src="https://cloud.githubusercontent.com/assets/3487289/21535260/4a2d57e0-cd27-11e6-910d-25eb770a085a.png">

This should be rendered instead as:

<img width="778" alt="screen shot 2016-12-28 at 18 00 36" src="https://cloud.githubusercontent.com/assets/3487289/21535283/8de54f1a-cd27-11e6-9db9-5670e9f9364a.png">

MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-12-28 18:00:10</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.13-dev-bd12c62</p>
            </projectDesc>
        </encodingDesc>
        <extMeta>
            <frames xmlns:humxml="http://www.humdrum.org/ns/humxml">
                <metaFrame n="17" token="!!!RDF**kern: j = place on staff below" xml:id="loc17">
                    <frameInfo>
                        <startTime float="8" />
                        <frameType>reference</frameType>
                        <referenceKey>RDF**kern</referenceKey>
                        <referenceValue>j = place on staff below</referenceValue>
                    </frameInfo>
                </metaFrame>
            </frames>
        </extMeta>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000117234565">
                        <staffGrp xml:id="m-000000189286276" symbol="brace" barthru="true">
                            <staffDef xml:id="staffdef-000000052169427" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5" />
                            <staffDef xml:id="staffdef-000000042576383" clef.shape="F" clef.line="4" meter.count="4" meter.unit="4" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-000000141954475">
                        <measure xml:id="measure-L4" n="1">
                            <staff xml:id="staff-L4F2" n="1">
                                <layer xml:id="layer-L4F2" n="1">
                                    <chord xml:id="chord-L5F2" dur="2">
                                        <note xml:id="note-L5F2S1" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L5F2S2" staff="2" oct="3" pname="f" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L6F2" dur="2">
                                        <note xml:id="note-L6F2S1" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L6F2S2" staff="2" oct="3" pname="e" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <staff xml:id="staff-L4F1" n="2">
                                <layer xml:id="layer-L4F1" n="1">
                                    <note xml:id="note-L5F1" dur="1" oct="2" pname="a" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L7" n="2" right="end">
                            <staff xml:id="staff-L7F2" n="1">
                                <layer xml:id="layer-L7F2" n="1">
                                    <beam xml:id="beam-000000024568933">
                                        <note xml:id="note-L9F2" dur="8" oct="5" pname="a" accid.ges="n" />
                                        <note xml:id="note-L10F2" dur="8" oct="5" pname="g" accid.ges="n" />
                                    </beam>
                                    <note xml:id="note-L11F2" dur="4" oct="5" pname="g" accid.ges="n" />
                                    <beam xml:id="beam-000000177647318">
                                        <note xml:id="note-L12F2" dur="8" oct="5" pname="a" accid.ges="n" />
                                        <note xml:id="note-L13F2" dur="8" oct="5" pname="f" accid.ges="n" />
                                    </beam>
                                    <note xml:id="note-L14F2" dur="4" oct="5" pname="e" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L9F3" n="2">
                                    <chord xml:id="chord-L9F3" dur="2" stem.dir="up">
                                        <note xml:id="note-L9F3S1" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L9F3S2" staff="2" oct="3" pname="f" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L12F3" dur="2" stem.dir="up">
                                        <note xml:id="note-L12F3S1" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L12F3S2" staff="2" oct="3" pname="e" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <staff xml:id="staff-L7F1" n="2">
                                <layer xml:id="layer-L7F1" n="1">
                                    <note xml:id="note-L9F1" dur="1" oct="3" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
Fixed 77095bb850f3ee2b0f8bac934966af9f88ac9ac5

![image](https://cloud.githubusercontent.com/assets/689412/25567333/770c4a80-2deb-11e7-8d0e-999a22241e67.png)
